### DISCUSSION TITLE
AddressType - definition as `0x{string}` -- is this better than `string`?

### DISCUSSION THREAD

**Discussion by appleseed-iii** [wagmi-dev/abitype#49] [General]
I'm curious on the thinking behind [defining AddressType](https://github.com/wagmi-dev/abitype/blob/7147b07decaa2f6071d33de688859bdac7ce4fdc/src/config.ts#L29) as `0x{string}`? It feels overly restrictive for very little increase in type safety. I guess I can't really come up with a big negative except now I need to go change all my `address: string` types 😭

**Comment by tmm** [comment]
I'd like to think so. That way you don't accidentally pass in a non-address value. It also makes types more obvious if you use ABIType in functions for autocomplete, etc.

That said, you can [configure ABIType](https://github.com/wagmi-dev/abitype#configuration) to use whatever `AddressType` you want:

```ts
declare module 'abitype' {
  export interface Config {
    AddressType: string
  }
}
```

**Reply by appleseed-iii** [reply]
good thinking 💪

**Reply by zkSoju** [reply]
Is the best approach to just assert the type using `as` when you know it is an address value? Or have a wrapper function?

**Reply by tmm** [reply]
You could assert. Ideally the `AddressType` would be a native type in your code (e.g. response from APIs, frontend state management, etc.).

**Comment by vseehausen** [comment]
The idea of checking if an Address starts with a 0x is great in theory, but in practice, it makes too many problems:

1. I usually import a JSON with contract addresses into my frontend repo.
2. These json imports don't get checked.
3. I could cast all types in a JS/TS file, but this would mean to much manual work when I redeploy.
4. I end up doing this often:

```
liquidityPoolContract()
			.balanceOf(/** @type {import('abitype').Address} Address */ ($account.address))
```

Which then renders the whole type-checking meaningless.

Also, most errors with addresses are based on a wrong address-chain connection and can only be found at runtime.

So, this library works great, but this one opinionated configuration could imho be changed. <3

**Reply by tmm** [reply]
Thanks for the info! Have you tried configuring the `AddressType`?

- **Discussion https://github.com/wagmi-dev/abitype/discussions/49#discussioncomment-4009201**
- **Docs https://abitype.dev/config.html**

**Reply by vseehausen** [reply]
Hey, that worked for the last project, but now I am use JSDoc only, which I did not manage to configure. I honestly think the 0x is not doing a typecheck, but a content check. I think this should be changed.

Opened this one: https://github.com/wevm/abitype/discussions/226
